x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/utils.ts

```diff
@@ -19,3 +23,32 @@ export const extractDomainAndEntryPointFromUrl = (
 
   return { domain, entryPoint };
 };
+
+export const getDomainWithProtocol = async (domain: string) => {
```

Comment:
[super optional] Any thoughts on making this a logic action/listener (even if it doesn't get called by anything other than another listener)? It's definitely minor, but it lets you keep your code together, so that someone doesn't have to open up a separate file to cross-reference when checking to see what it actually does.

FWIW for testing, you could still mock/stub this helper via `jest.spyOn(AddDomainLogic.actions, 'getDomainWithProtocol').mockImplementation()`.

No worries either way though! Just a thought

Reply:
After having fully reviewed the PR and coming back to this initial comment, I'm now super 50/50 on my own suggestion. I think it depends on how large/bloated AddDomainLogic is. If it has a lot going on as it is, a separate utils file so that the logic file and its tests don't get ovelry large feels like a better dev experience to me.

TL;DR ignore me lol

Reply:
Something I think worth discussing after the migration